src/frequenz/client/reporting/_client.py

```diff
@@ -299,7 +299,13 @@ def dt2ts(dt: datetime) -> PBTimestamp:
 
         stream_filter = PBReceiveMicrogridComponentsDataStreamRequest.StreamFilter(
             time_filter=time_filter,
-            resampling_options=PBResamplingOptions(resolution=resolution),
+            resampling_options=PBResamplingOptions(
+                resolution=(
+                    int(resampling_period.total_seconds())
```

Comment:
I am wondering, do we want `int()` which cuts off the fractional part or rather `round()` which .. well.. rounds ;)

Reply:
I am not sure, but we had resolution : int beforehand, that's why I kept it this way

Reply:
But it also makes sense, cause the client/user would likely use 1min, 15min and probably nothing too complicated in between. If it needs to be as it is read out then one can just leave it defaulting to None.

Reply:
sure, but due to floating point, you could end up with an number like 4.999999 in there and instead of 5 you'll get 4 this way.

Reply:
But if the proto defines `uint32` should we then not also use `int` here?

Reply:
You would still use int because the API doesn't support sub-second resolution. But int just cuts off the fractional part, so what @Marenz describes is a problem with floating points in general where you cannot always represent integers exactly. So I agree with rounding here.

Reply:
>  where you cannot always represent integers exactly.

In practice this should less of a concern in the range we use this parameter, maybe more if the value is a result of floating point operations I guess. But it's still good practice :)